bitsail-components/bitsail-component-metrics/bitsail-component-metrics-prometheus/src/main/java/com/bytedance/bitsail/component/metrics/prometheus/impl/PrometheusPushGatewayReporter.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2022 Bytedance Ltd. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.bytedance.bitsail.component.metrics.prometheus.impl;
+
+import com.bytedance.bitsail.base.metrics.Scheduled;
+import com.bytedance.bitsail.common.configuration.BitSailConfiguration;
+import com.bytedance.bitsail.component.metrics.prometheus.AbstractPrometheusReporter;
+import com.bytedance.bitsail.component.metrics.prometheus.option.PrometheusPushGatewayOptions;
+
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.dropwizard.DropwizardExports;
+import io.prometheus.client.exporter.PushGateway;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class PrometheusPushGatewayReporter extends AbstractPrometheusReporter implements Scheduled {
+  private int periodSeconds;
+  private PushGateway pushGateway;
+  private String jobName;
+  private boolean deleteOnShutdown;
+  DropwizardExports dropwizardExports;
+
+  @Override
+  public void open(BitSailConfiguration configuration) {
+    String serverHost = configuration.get(PrometheusPushGatewayOptions.PUSHGATEWAY_HOST_NAME);
+    int serverPort = configuration.get(PrometheusPushGatewayOptions.PUSHGATEWAY_PORT_NUM);
+    periodSeconds = configuration.get(PrometheusPushGatewayOptions.PUSHGATEWAY_REPORT_PERIOD_IN_SECONDS);
+    jobName = configuration.get(PrometheusPushGatewayOptions.PUSHGATEWAY_JOBNAME);
+    deleteOnShutdown = configuration.get(PrometheusPushGatewayOptions.PUSHGATEWAY_DELETE_ON_SHUTDOWN_ENABLE);
+    pushGateway = createPushGatewayClient(serverHost, serverPort);
+    dropwizardExports = new DropwizardExports(metricRegistry);
+    dropwizardExports.register(CollectorRegistry.defaultRegistry);
+  }
+
+  @SuppressWarnings("checkstyle:MagicNumber")
+  private PushGateway createPushGatewayClient(String serverHost, int serverPort) {
+    if (serverPort == 443) {
+      try {
+        return new PushGateway(new URL("https://" + serverHost + ":" + serverPort));
+      } catch (MalformedURLException e) {
+        e.printStackTrace();
```

Comment:
you can log the stack trace other than print

Reply:
The latest version use BitSailException to deal with it, the reason that I use exception but not use log is I don't want let user find they initialize the prometheus report fail until they finish the job, they should make sure the monitoring system work when they start the job.
![image](https://user-images.githubusercontent.com/47937942/217168889-cda89c8d-7967-4ad8-b48b-0788547567b0.png)
![image](https://user-images.githubusercontent.com/47937942/217168836-0e572e9a-3017-4598-b4b6-508ea582822e.png)